circle.yml

```diff
@@ -0,0 +1,29 @@
+machine:
+  python:
+    version: 2.7.11
+  services:
+    - docker
+
+dependencies:
+  override:
+    - bash .ci/install.sh
+    - python .ci/prepare.py
+
+
+compile:
+  override:
+    - bash .ci/build_images.sh:
+        parallel: true
+
+
+test:
+  override:
+    - bash .ci/test_images.sh:
+        parallel: true
+
+
+deployment:
```

Comment:
Yes, deployment only happens after all tests have passed I believe, that's why I think it's better to push the images at deployment stage instead of test stage.

Deployment stage only runs on one VM worker, so I think if any one test fails, no deployment will happen. This is my theory right now, we will know how it actually works when I start working on this stage.